src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    private static class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    static int hash(Object key) {
+        return (key == null) ? 0 : key.hashCode() & 1 << 16;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (table == null) {
+            resize();
+        }
+        Node<K,V> node = new Node<>(hash(key), key, value, null);
 
+        int index;
+        if (table[(index = hash(key) % table.length)] == null) {
+            table[index] = node;
```

Comment:
Why do you find index here if you have  hash  method  for it , it's hardcode ,let's don't repeat yourself

Reply:
hash method calculate a hash for key. Using it when create new Node